### ISSUE TITLE
Installing fails with permission checks

### ISSUE THREAD

**Issue by iBotPeaches** [TryGhost/Ghost-CLI#388]
### This issue is a

* [x] Bug Report
* [x] Feature Request

### Summary

Installing into `/home/user/ghost10` directory where `user` has full control over everything in that user directory (`home/user`) fails.

### Steps to Reproduce (for a bug report)

1. Install `ghost-cli` over sudo.
2. `su connor`
(where connor owns the entire `/home/connor` directory)
3. `mkdir ghost10; cd ghost10;`
4. `ghost install -V`

```
connor@connortumbleson.com [~/ghost10]# pwd
/home/connor/ghost10
connor@connortumbleson.com [~/ghost10]# ghost install -V
[12:38:43] Checking system Node.js version [started]
[12:38:43] Checking system Node.js version [completed]
[12:38:43] Checking current folder permissions [started]
[12:38:43] Checking current folder permissions [failed]
[12:38:43] → The path /home/connor/ is not readable by other users on the system.
This can cause issues with the CLI, please either make this directory readable by others or install in another location.
A SystemError occured.
```

I see this commit was added to check the entire path back to root for readability - https://github.com/TryGhost/Ghost-CLI/commit/aa5b047064c616fc87c02a410f53b19c4022337e

Perhaps I don't understand, but my user has full permission to his entire user directory. Any other users on the system do not, with exception of root. However, reading the documentation: https://docs.ghost.org/v1.0.0/docs/cli-knowledge-base#section-permissions

It says ghost-cli makes its own user (`ghost`) - while I wish it could just use the already restricted user I provide it (`connor`). I just don't understand why it cares about directories outside of the current selected install directory. 

Can I tell ghost-cli to use the user I specify? There are multiple ghost installations already on this server and they follow the `/home/user` pattern where `user` owns the ghost process running. I'd rather not switch that method to all ghost blogs running from the same user.

Apologies if this is user error and not bug report worthy.

### Technical details

Debug Information:
    Node Version: v6.9.4
    Ghost-CLI Version: 1.0.0
    Environment: production
    Command: 'ghost install -V'

**Comment by acburdine** [comment]
@iBotPeaches I see what you're getting at here - will definitely leave open for future improvements in this area.

One thing I would want to note though - the restricted user you have must also have sudo permissions if you want the CLI to auto-configure things like nginx and systemd. This is the reason why the specific `ghost` user was created - because the CLI needs certain sudo permissions it auto-creates a locked-down user.

Would potentially be open as well to being able to tell the CLI what to name the specific, locked-down user, that way you could do something like `ghost install --user connor` and ghost would use your locked down user rather than create a new `ghost` one.

**Comment by ErisDS** [comment]
I think this is a major edge case, and I would be in favour of closing it unless we get more requests for it.

You can create the `ghost` user yourself before running Ghost CLI and then Ghost CLI will use the existing user instead of creating one. However, then you have to make sure you get the permissions correct.

> I just don't understand why it cares about directories outside of the current selected install directory.

It cares about the directories _above_ not outside of the current directory and this is because of how permissions cascade in linux.

**Comment by iBotPeaches** [comment]
Thanks for quick responses. I'll go about with making `ghost` user and giving necessary permissions in order to resolve this. As you said, probably edge case. So you can close this to keep tracker clean.

**Comment by ErisDS** [comment]
Cool - let's see how this plays out 🙂

**Comment by crunchprank** [comment]
Hi @ErisDS I thought I would jump in here and say that I also run my production server similar to @iBotPeaches:

> There are multiple ghost installations already on this server and they follow the /home/user pattern where user owns the ghost process running

My personal use case is because it's a shared hosting environment and I'd prefer each blog to be ran by its corresponding user. So for what it's worth, that's my input. 🙂

**Comment by Blindmikey** [comment]
Running into this issue.

**Comment by dinosaw** [comment]
I know this is tagged as investigating but it is also closed. Are there any plans on implementing a way around this? My setup is similar to OP as well.

**Comment by crunchprank** [comment]
@dinosaw I don't know if this is a "solution" or whether or not the Ghost team recommends this being used in a production environment, but I recently saw a comment in #405 (https://github.com/TryGhost/Ghost-CLI/pull/409) where @acburdine added the following option when installing: `--no-setup-linux-user`. I played around with using this and it fixed the issues I was running into, though I haven't been able to test it extensively. For now, seems to get me where I wanted though.
